hs_core/serialization.py

```diff
@@ -0,0 +1,1061 @@
+
+    for fp in file_paths:
+        fname = os.path.basename(fp)
+        fsize = os.stat(fp).st_size
+        fd = open(fp, 'rb')
+        res_files.append(UploadedFile(file=fd, name=fname, size=fsize))
+
+    return res_files
+
+
+def create_resource_from_bag(bag_content_path, preserve_uuid=True):
+    """
+    Create resource from existing uncompressed BagIt archive.
+
+    Follows roughly the same pattern as hs_core.views.create_resource().
+
+    :param bag_content_path:
+    :return: None if successful.  If the resource associated with the bag
+    depends upon the existence of another resource in order for the dependant
+    resource's metadata to be fully created, this function will return a
+    Tuple<String, ResourceMeta, Resource>), where String is a string
+    representing  the resource ID that the Bagged Resource depends on,
+    ResourceMeta is a subclass of hs_core.serialization.GenericResourceMeta
+    that represents the metadata of the newly created resource, and Resource
+    is a subclass of hs_core.model.BaseResource representing the newly created
+    resource. Once the dependant resource has been created, the caller is
+    responsible for calling ResourceMeta.write_metadata_to_resource(Resource)
+    to ensure that all metadata fields have been populated.
+
+    :raises: HsDeserializationException if an error occurred.
+    """
+    # Get resource metadata
+    resource_files = None
+    rm = None
+    try:
+        rm = GenericResourceMeta.read_metadata_from_resource_bag(bag_content_path)
+        # Filter resource files
+        resource_files_filtered = []
+        for f in rm.files:
+            if rm.include_resource_file(f):
+                resource_files_filtered.append(f)
+        res_files = [os.path.join(bag_content_path, f) for f in resource_files_filtered]
+        resource_files = _prepare_resource_files_for_creation(res_files)
+
+    except GenericResourceMeta.ResourceMetaException as e:
+        msg = "Error occurred while trying to create resource from bag path {0}. "
+        msg += "Error was: {1}"
+        msg = msg.format(bag_content_path, str(e))
+        raise HsDeserializationException(msg)
+
+    # Send pre-create resource signal
+    try:
+        if resource_files is None:
+            resource_files = []
+
+        page_url_dict, res_title, metadata = resource_pre_create_actions(resource_type=rm.res_type,
+                                                                         files=resource_files,
+                                                                         resource_title=rm.title,
+                                                                         page_redirect_url_key=None)
+    except ResourceFileSizeException as ex:
+        raise HsDeserializationException(ex.message)
+
+    except ResourceFileValidationException as ex:
+        raise HsDeserializationException(ex.message)
+
+    except Exception as ex:
+        raise HsDeserializationException(ex.message)
```

Comment:
This could really mask errors that happen underneath, making bugs harder to track down.